iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/RevertConnectVoltageLevelOnLine.java

```diff
@@ -0,0 +1,194 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.modification.topology;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.commons.reporter.Reporter;
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.iidm.modification.NetworkModification;
+import com.powsybl.iidm.network.Branch;
+import com.powsybl.iidm.network.Line;
+import com.powsybl.iidm.network.LineAdder;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.VoltageLevel;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import static com.powsybl.iidm.modification.topology.ModificationReports.createdLineReport;
+import static com.powsybl.iidm.modification.topology.ModificationReports.noVoltageLevelInCommonReport;
+import static com.powsybl.iidm.modification.topology.ModificationReports.notFoundLineReport;
+import static com.powsybl.iidm.modification.topology.ModificationReports.removedLineReport;
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.LoadingLimitsBags;
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.addLoadingLimits;
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.attachLine;
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.calcMinLoadingLimitsBags;
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.createLineAdder;
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.removeVoltageLevelAndSubstation;
+
+/**
+ * This method reverses the action done in the ConnectVoltageLevelOnLine class :
+ * it replaces 2 existing lines (with the same voltage level at one of their side) with a new line,
+ * and eventually removes the voltage level in common (attachment point), if it contains no equipments anymore, except bus or bus bar section
+ *
+ *    VL1 ----------- attachment point ----------- VL2         =========>        VL1 ------------------------- VL2
+ *          (line1)                      (line2)                                              (line)
+ *
+ *
+ * @author Franck Lecuyer <franck.lecuyer at rte-france.com>
+ */
+public class RevertConnectVoltageLevelOnLine implements NetworkModification {
+
+    private String line1Id;
+    private String line2Id;
+
+    private String lineId;
+    private String lineName;
+
+    /**
+     * Constructor.
+     *
+     * @param line1Id       The non-null ID of the first line
+     * @param line2Id       The non-null ID of the second line
+     * @param lineId        The non-null ID of the new line to be created
+     * @param lineName      The optional name of the new line to be created
+     */
+    public RevertConnectVoltageLevelOnLine(String line1Id, String line2Id, String lineId, String lineName) {
```

Comment:
The constructors should be private-package as only the builder should be used

Reply:
Changed